.github/workflows/master-publish-docker-image.yaml

```diff
@@ -51,27 +51,52 @@ jobs:
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
 
-      - name: 'Setup gcloud auth'
-        id: 'gcloud-auth'
-        uses: 'google-github-actions/auth@v2'
+  helm-deploy:
+    environment: master
+    needs: build-and-push-image
+    runs-on: helsinki-crawler-proxy-production-arc-runners
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
         with:
-          project_id: ${{ secrets.GCLOUD_PROJECT_ID }}
-          workload_identity_provider: ${{ secrets.GCLOUD_WORKER_IDENTITY_PROVIDER }}
-
-      - name: "Setup gcloud CLI" 
-        id: 'gcloud-setup'
-        uses: 'google-github-actions/setup-gcloud@v2'
-
-      - name: "Decrypt SSH key"
+          ref: master
+      - name: Install ARC dependencies
         run: |
-          echo "${{ secrets.GCLOUD_SSH_PRIVATE_KEY }}" | base64 --decode > key && 
-          echo "${{ secrets.GCLOUD_SSH_PUBLIC_KEY }}" | base64 --decode > key.pub && 
-          chmod 400 key && 
-          chmod 400 key.pub
-
-      - name: 'Set up gcloud CLI'
-        uses: 'google-github-actions/setup-gcloud@v2'
-
-      - name: 'Update deployment'
-        run: 'gcloud compute ssh ${{ secrets.GCLOUD_USER }}@helsinki-crawler-proxy --zone ${{ secrets.GCLOUD_ZONE }} --project=${{ secrets.GCLOUD_PROJECT_ID }} --ssh-key-file=key --tunnel-through-iap --command "/home/${{ secrets.GCLOUD_USER }}/update/update.sh ${{ github.ref_name }} production ${{ steps.dockerpush.outputs.digest }}"'
-        
+          sudo apt update && sudo apt install -y zip
+      - name: Import Secrets
+        id: import-secrets
+        uses: hashicorp/vault-action@v2
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          token: ${{ secrets.VAULT_TOKEN }}
+          secrets: |
+              ${{ secrets.VAULT_PATH }} HTTP_PORT | HTTP_PORT ;
+              ${{ secrets.VAULT_PATH }} HTTPS_PORT | HTTPS_PORT ;
+      - name: Install Helm
+        uses: azure/setup-helm@v3
+      - name: Set up kubectl
+        uses: azure/setup-kubectl@v1
+      - name: Install Chart 
+        run: |
+            echo "replicaCount: 1" >> new-values.yaml &&
```

Comment:
just 1 replica is not correct (I believe it should be 3). It should also originate from HCV